cas-server-core-web/src/main/java/org/jasig/cas/web/flow/AbstractNonInteractiveCredentialsAction.java

```diff
@@ -68,17 +78,21 @@ protected final Event doExecute(final RequestContext context) {
         final String ticketGrantingTicketId = WebUtils.getTicketGrantingTicketId(context);
         final Service service = WebUtils.getService(context);
 
-        if (isRenewPresent(context)
-            && ticketGrantingTicketId != null
-            && service != null) {
+        if (isRenewPresent(context) && ticketGrantingTicketId != null && service != null) {
 
             try {
+                final AuthenticationContextBuilder builder = new DefaultAuthenticationContextBuilder(
```

Comment:
Of course, my counter proposal could not be the final source code, but it may help to find improvments. Even for me, it's hard to read the changes and I've spent quite some time on this PR...